### ISSUE TITLE
📚[DOC]: CorrDiff Validation and Early Stopping

### ISSUE THREAD

**Issue by maxfield-green** [NVIDIA/physicsnemo#860]
### How would you describe the priority of this documentation request

Medium

### Is this for new documentation, or an update to existing docs?

Update

### Describe the incorrect/future/missing documentation

In CorrDiffs ```train.py``` there is support for logging validation loss based on the presense of a validation key in the training config yaml. In [dataset.py](https://github.com/NVIDIA/physicsnemo/blob/ad863fdb8ff2e263551e4181b7799f79a4c5177d/examples/generative/corrdiff/datasets/dataset.py#L121) it looks like the training dataset configuration is copied for repurposing for validation with the option to update the configuration. I can see how one would provide a secondary path to a separate dataset used for validation, which would replace the dataset.data_path in either the training config (https://github.com/climacell/physicsnemo/blob/main/examples/generative/corrdiff/conf/config_training_hrrr_mini_regression.yaml) or via hydra override. Is this the intended validation process? I am confused by the```test_train_split``` boolean used to initiate the validation flow if the validation is done on a separate dataset and not actaully split out from the training dataset. 

Additionally, if validation logging is supported, is there configurable early stopping for the regression? 

Thanks for clarity.

**Comment by CharlelieLrt** [comment]
@maxfield-green thank you for your question.

TL;DR: 
1. You should provide a `validation` section in your config file and put there any field that you might want to overwrite [_in comparison to the training dataset_ config].
2. No built-in early stopping, but possibility to do it _a posteriori_.

### Validation

For example, let's say you have a training dataset config that looks like:

```yaml
dataset:
    data_path: <path_to_training_dataset>
    stats_path: <path_to_stats_file>
    output_variables: ["u10m", "v10m", "t2m", "precip"]
    # [Other fields]
```

If you want to do do validation, say on the same dataset but with different output variables (**note:** this is only hypothetical, I am only trying to illustrate the logic here), you should do:

```yaml
validation:
    output_variables: ["u10m", "v10m"]  # Overwrites the field provided in dataset
    # [Other fields to overwrite]
```

This will create a validation dataset with the same `data_path` and `stats_path` as the training set, but with a different set of `output_variables`.

### Early-stopping

Regarding early-stopping, we currently have no built-in configurable way to do it, but the `train.py` should save multiple checkpoints and log the associated losses, so you just need to look at those to pick _a posteriori_ the best model.

**Comment by maxfield-green** [comment]
Got it, thank you for clarifying! Maybe recommend replacing the ```test-train-split``` boolean with ```valdiation``` to reduce confusion?

**Comment by CharlelieLrt** [comment]
Thank you for the suggestion. We will keep this issue open until we do the renaming.
